### ISSUE TITLE
Add Alter schema|Table  statements to dask-sql

### ISSUE THREAD

**Issue by rajagurunath** [dask-contrib/dask-sql#213]
These Alter commands are needed in software like **dbt** for creating views and renaming those entities etc.
so probably it will be helpful to have the following statements 

1. ALTER SCHEMA
2. ALTER TABLE 
 
For ALTER TABLE, we may use following presto docs as reference which contains (`rename`,`add column`,`drop column`)
  https://prestodb.io/docs/current/sql/alter-table.html
 
 
 ```
ALTER TABLE [ IF EXISTS ] name RENAME TO new_name
ALTER TABLE [ IF EXISTS ] name ADD COLUMN [ IF NOT EXISTS ] column_name data_type [ COMMENT comment ] [ WITH ( property_name = expression [, ...] ) ]
ALTER TABLE [ IF EXISTS ] name DROP COLUMN column_name
ALTER TABLE [ IF EXISTS ] name RENAME COLUMN [ IF EXISTS ] column_name TO new_column_name
```

**Comment by rajagurunath** [comment]
Hi @nils-braun,

I tried adding Alter schema command in a couple of ways ( inheriting from `SqlAlter` and `SqlCall`) for the last few days but didn't get any luck.

Error : ```  dask_sql.utils.ParsingException: Non-query expression encountered in illegal context ```

I am suspecting this issue is from the Java Calcite SQL parsing side, I am  Beginner in both Java and Calcite, So need your help to sort this issue, (I think we need to do some changes in `RelationalAlgebraGenerator.java` to make calcite recognize `alter statements` what do you think? 🤔 ) 

Please review the below code and let me know where I am doing wrong.

https://github.com/dask-contrib/dask-sql/compare/main...rajagurunath:feature/alter_schema

Thanks

**Comment by nils-braun** [comment]
Hi @rajagurunath - sure!
The parsing logic of Calcite (or better: the templating logic and javaCC) is still also a mystery to me... (and this is actually not the only occurrence of such a problem in the last days, see https://github.com/dask-contrib/dask-sql/pull/228#pullrequestreview-740364960)

You can fix the immediate problem by removing "ALTER" from the list of `nonReservedKeywordsToAdd`. However, the problem then is that the `ALTER` statements in Calcite always need a scope, which is either "SESSION" or "SYSTEM". So the SQL queries `ALTER SESSION SCHEMA "test" RENAME TO "test2"` will be correct now, but not `ALTER SCHEMA "test" RENAME TO "test2"` (and I guess this is what you want).

So I would propose to do the following: Rewrite the `SqlAlterSchema` to also include the `ALTER` command, something like
```

SqlNode SqlAlterSchema() :
{
    final Span s;
    final SqlIdentifier oldSchemaName;
    final SqlIdentifier newSchemaName;
}
{
    <ALTER>
    { s = span(); }
    <SCHEMA>
    oldSchemaName = SimpleIdentifier()
    <RENAME> <TO>
    newSchemaName = SimpleIdentifier()
    {
         return new SqlAlterSchema(s.end(this), oldSchemaName, newSchemaName);
    }
}
```
and move it to `statementParserMethods` (and still remove `ALTER` from the `nonReservedKeywordsToAdd`). That should do it (at least worked for me).

Hope this helps.

**Comment by charlesbluca** [comment]
Closed with #285
